e3sm_diags/driver/utils/regrid.py

```diff
@@ -526,7 +526,8 @@ def _hybrid_to_plevs(
             method="log",
             target_data=pressure_coords,
         )
-
+    # Keep units unchanged.
+    result[var_key].attrs["units"] = ds[var_key].units
```

Comment:
somehow the units are not preserved with vertical regridding, not sure if its `xgcm` or  and `xcdat` problem.

Reply:
Based on this xcdat [code](https://github.com/xCDAT/xcdat/blob/fbf1db68fd6328d5a534e0c1e57e22e6fdd70c76/xcdat/regridder/xgcm.py#L200-L208), `xcdat` _should_ be preserving attributes (while `xgcm` does not), 

I added `xr.set_options(keep_attrs=True)` to this code block just to make sure. The unit tests for `regrid_z_axis_to_plevs()` also check to make sure attributes are preserved/updated ([example](https://github.com/E3SM-Project/e3sm_diags/blob/1530283382ac2012c77d820a2f5d24d0df5e7618/tests/e3sm_diags/driver/utils/test_regrid.py#L329-L353)). 

I'm not 100% sure if this line is needed here. I'll step through the code to check.

Reply:
I stepped through the code and found that the units _are_ correctly being preserved in xCDAT. Since the variable is being regridded from hybrid to pressure levels, the units of the pressure level is being used ("mb").

Are we supported to maintain the original units of the variable? The old CDAT version of this function also sets "mb" as the variable's units:

https://github.com/E3SM-Project/e3sm_diags/blob/a6bbc1c893ff81c480cc0793409f4889f7f76d5d/e3sm_diags/driver/utils/general.py#L131-L137

Reply:
Yes, we should maintain the original units of the variable. The CDAT version of the function is correct, it sets `mb` (milibar) as the units of the reconstructed 3d pressure. The corresponding refactored code is here https://github.com/E3SM-Project/e3sm_diags/blob/b1183a7375f058695d842814bef2c55ace55b480/e3sm_diags/driver/utils/regrid.py#L534-L586: 
The refactored code correctly set `mb` for pressure.

The units of the variable should be preserved here and also in the next vertical interpolation operation.

Reply:
Got it. We can keep your changes here and I will mark this thread as a resolved.